tests/pos/i828.scala

```diff
@@ -0,0 +1,3 @@
+object X {
+  val x: Int = null.asInstanceOf[Nothing]
```

Comment:
You should have test cases for the various nodes modified by `BottomTypes` and test cases for `Null` too.